ovn_k8s/modes/overlay.py

```diff
@@ -436,3 +466,366 @@ def sync_services(self, services):
+        from_clause = pseudo_acl[2]
+        if from_clause and from_clause != '*':
+            # '*' means every address matches and therefore no source IP
+            # match should be added to the ACL
+            src_match = r"ip4.src\=\=\{%s\}" % from_clause
+        match_items = set()
+        protocol_port_map = {}
+        for (protocol, port) in ports_clause:
+            ports = protocol_port_map.setdefault(protocol, set())
+            if port:
+                ports.add(port)
+        for protocol, ports in protocol_port_map.items():
+            if ports:
+                item_match_str = (r"%s.dst\=\=\{%s\}" %
+                                  (protocol.lower(), "\,".join(
+                                      [str(port) for port in ports
+                                       if port is not None])))
+            else:
+                item_match_str = protocol.lower()
+            match_items.add(item_match_str)
+        ports_match = "\ ||\ ".join(match_items)
+        if ports_match and src_match:
+            policy_match = "\(%s\)\ &&\ \(%s\)" % (ports_match, src_match)
+        elif ports_match:
+            policy_match = ports_match
+        elif src_match:
+            policy_match = src_match
+        else:
+            policy_match = None
+        outport_match = r'outport\=\=\"%s\"\ &&\ ip' % lport_name
+        if policy_match:
+            match = r'%s\ &&\ %s' % (outport_match, policy_match)
+        else:
+            match = outport_match
+        return match
+
+    def _policy_pod_traffic(self, pod_data, pseudo_acls):
+        pod_name = pod_data['metadata']['name']
+        pod_ns = pod_data['metadata']['namespace']
+        pod_id = pod_data['metadata']['uid']
+        # TODO(salv-orlando): do not remove the default drop in the next step
+        # and ensure the same rule is not created if already defined in the
+        # subsequent step. The current code will leave a pod without security
+        # rules or with incomplete security rules if an error occurs.
+        self.remove_acls(pod_id, remove_default=True)
+        # TODO(salv-orlando): lswitch & lport can be easily cached
+        lport_name = self._build_logical_port_name(pod_ns, pod_name)
+        lport_uuid = ovn_nbctl("--data=bare", "--no-heading",
+                               "--columns=_uuid", "find",
+                               "logical_switch_port",
+                               "name=%s" % lport_name)
+        lswitch_name = ovn_nbctl("--data=bare", "--no-heading",
+                                 "--columns=name", "find",
+                                 "logical_switch",
+                                 "ports{>=}%s" % lport_uuid)
+        for pseudo_acl in pseudo_acls:
+            match = self._create_acl_match(lport_name, pseudo_acl)
+            vlog.dbg("Pod: %s (Namespace:%s): ACL match: %s" % (
+                pod_name, pod_ns, match))
+            ovn_acl_data = (pseudo_acl[0], match, pseudo_acl[3])
```

Comment:
I've been using the tuple out of laziness. I could just use a named tuple indeed.